modin/backends/base/query_compiler.py

```diff
@@ -573,21 +573,57 @@ def min(self, **kwargs):
         """
         return DataFrameDefault.register(pandas.DataFrame.min)(self, **kwargs)
 
-    def prod(self, **kwargs):
-        """Returns the product of each numerical column or row.
+    def sum(self, squeeze_self, axis, **kwargs):
+        """Returns the sum of each numerical column or row.
 
         Return:
-            Pandas series with the product of each numerical column or row.
+            Pandas series with the sum of each numerical column or row.
         """
-        return DataFrameDefault.register(pandas.DataFrame.prod)(self, **kwargs)
+        # TODO: rework to original implementation after pandas issue #41074 resolves if possible.
+        def map_func(df, **kwargs):
+            if squeeze_self:
+                result = df.squeeze(axis=1).sum(**kwargs)
+                if is_scalar(result):
+                    if axis:
+                        return pandas.DataFrame(
+                            [result], index=["__reduced__"], columns=["__reduced__"]
+                        )
+                    else:
+                        return pandas.Series([result], index=[df.columns[0]])
+                else:
+                    return result
+            else:
+                return df.sum(**kwargs)
```

Comment:
I see functions like this in 6 places (`sum` and `prod` here, and `sum`/`prod`/`sum_min_count`/`prod_min_count` in pandas query compiler), can we reduce the amount of copy-pasteness somehow?

Reply:
All our map functions look similarly. The difference is that we register different functions.